build-script.py

```diff
@@ -250,6 +236,54 @@ def generate_gyb_files(gyb_exec, verbose, add_source_locations, destination=None
     print("Done Generating gyb Files")
 
 
+def make_dir_if_needed(path):
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+
+def clear_gyb_files_from_previous_run(sources_dir, destination_dir):
```

Comment:
Could you add a comment here like. It took me a while to understand what this is actually doing even though I was probably who wrote the code snippet.

```
# Remove any files in the `gyb_generated` directory that no longer have a corresponding `.gyb` file in the `Sources` directory.
```